WalletWasabi.Documentation/Productivity.md

```diff
@@ -0,0 +1,111 @@
+# Productivity
+
+Developers use to work with many branches at the same time because during any single day they review others' pull requests, try new crazy things, work on new features/bug-fixes and also fix somebody elses' PRs. All this forces a combination of `git stash [pop|save]`, `git fetch`, `git checkout [-f]` and again `git stash [pop|apply]`. No matter if this is done by command line or with a tool the problem of having only one copy of the code is a ugly constrain.
+
+## Wasabi Wallet folders organization (suggestion)
+
+### Create main local bare repository
+
+First clone your existing repository as a bare repo (a repository without files) and do it in a `.git` directory.
+
+```
+$ git clone --bare git@github.com:zkSNACKs/WalletWasabi.git .git
+```
+
+### Add devs remote repositories
+
+Having the team members' repositories as remote makes the work easier: 
```

Comment:
In addition to named branches, it's also useful to [use the `gh` tool](https://cli.github.com/manual/gh_pr_checkout) like @kiminuo reccomends, or add a fetch spec for numbered pull request refs, e.g.:

```diff
 [remote "origin"]
 	url = https://github.com/zkSNACKs/WalletWasabi.git
 	fetch = +refs/heads/*:refs/remotes/origin/*
+	fetch = +refs/pull/*/head:refs/remotes/origin/pull/*
```